articles/openshift/oshift-how-apply-security-best-practices.md

```diff
@@ -0,0 +1,70 @@
+
+- Limiting Resources - Containers should always be set up with resources limits, in the case of a denial of service attack or just a burst of very high load on a service these resource limits will prevent pods from using all available resources on the node they are running on. You can read more about setting resource quotas and limits [here](https://docs.openshift.com/container-platform/3.11/dev_guide/compute_resources.html)
+
+- Privileged ports - TCP/IP ports below port 1024 are considered to be Privileged Ports and should not be used to front any Service that is deployed into OpenShift.
+
+- Route security - Within OpenShift users can expose their services using routes. Creating a route to a service allows the user to access the service through a URL on a browser. By default, the URLS which are created are exposed on HTTP port 80 without any SSL certificate. Users can secure these routes with SSL encryption using one of three SSL termination methods:
+     1. Edge - SSL termination is done on the HAProxy router from which point onwards the traffic is un-encrypted.
+     2. Passthrough termination - No SSL de-Cryption on the routers instead connections are passed straight through the routers to the OpenShift apps which are then responsible for SSL certificate termination and traffic decryption.
+     3. Re-Encrypt - This process uses 2 certificates, first certificate is held on HAProxy router and the second in the pods. Whereby securing the traffic all the way down to the application.
+
+It is possible to further securing routes by applying IP Whitelisting on routes, we have documented this process here [IP Whitelisting](https://docs.ukcloud.com/articles/openshift/oshift-how-restrict-access-to-openshift-routes-by-ip-address.html).
+
+UKCloud can whitelist source IP's/Subnets in OpenShift to the OpenShift GUI/Metrics page and the services that customers deploy which are exposed on various community networks with the use of security groups. These restrictions can be applied at the time cluster deployment or later through a service request to UKCloud OpenShift team.
+
+- OpenShift uses a Software-Defined Networking (SDN) approach to provide cluster wide networking that enables communication between pods across the OpenShift cluster. This Container Networking Interface (CNI) runs in the 'OpenShift SDN' project and creates an overlay network using the Open vSwitch (OVS) networking plugin. However, the main thing to note here is that UKCloud will be deploying OpenShift clusters as of February 2019 with the ovs-networkpolicy plugin. This plugin allows project administrators to configure their own isolation policies using [NetworkPolicy objects](https://docs.openshift.com/container-platform/3.6/admin_guide/managing_networking.html#admin-guide-networking-networkpolicy).
```

Comment:
OpenShift uses a software-defined networking (SDN) approach to provide cluster-wide networking that enables communication between pods across the OpenShift cluster. This Container Networking Interface (CNI) runs in the OpenShift SDN project and creates an overlay network using the Open vSwitch (OVS) networking plugin.

However, from February 2019, UKCloud will be deploying OpenShift clusters with the ovs-networkpolicy plugin. This plugin allows project administrators to configure their own isolation policies using [NetworkPolicy objects](https://docs.openshift.com/container-platform/3.6/admin_guide/managing_networking.html#admin-guide-networking-networkpolicy).